tests/libvmi/factory.go

```diff
@@ -34,14 +34,30 @@ const (
 // NewFedora instantiates a new Fedora based VMI configuration,
 // building its extra properties based on the specified With* options.
 func NewFedora(opts ...Option) *kvirtv1.VirtualMachineInstance {
-	return newFedora(cd.ContainerDiskFedora, opts...)
+	return newFedoraWithUserDataSetPassword(cd.ContainerDiskFedora, opts...)
 }
 
 // NewTestToolingFedora instantiates a new Fedora based VMI configuration,
 // building its extra properties based on the specified With* options.
 // This image has tooling for the guest agent, stress, and more
 func NewTestToolingFedora(opts ...Option) *kvirtv1.VirtualMachineInstance {
-	return newFedora(cd.ContainerDiskFedoraTestTooling, opts...)
+	return newFedoraWithUserDataSetPassword(cd.ContainerDiskFedoraTestTooling, opts...)
+}
+
+// newFedoraWithUserDataSetPassword instantiates a new Fedora based VMI configuration,
+// building its extra properties based on the specified With* options.
+// Adding to it UserData which sets the password
+func newFedoraWithUserDataSetPassword(containerDisk cd.ContainerDisk, opts ...Option) *kvirtv1.VirtualMachineInstance {
+	userDataSetPassword := `#!/bin/bash
+	echo "fedora" | passwd fedora --stdin
+	echo `
+
+	fedoraOptions := []Option{
```

Comment:
not sure I'm getting it.

the commit message suggests you're removing the userDataSetPassword
 from the configuration, but from what I understand, you've separated it to a different function `newFedoraWithUserDataSetPassword` and then send that to use that to `newFedora`. 
So eventually you **do** add it, but in a former function.. IIUC nothing changed really.. 
Am I missing something?

Reply:
I removed it only from NewSriovFedora, the change you point to is to keep the non sriov intact

Reply:
yep you're right, must have missed it. then I'm good. thanks

Reply:
Can you also explain why this is an issue only with NewSriovFedora vms and this race does not also occur on regular ones?

Reply:
the race do exists on the regular ones, the solution there atm is different 
https://github.com/kubevirt/kubevirt/pull/5019
It disable the Guest Agent service
install it as the last thing it does on Cloud-Init, after setting the password on Cloud-Init,
wait for the Guest Agent connected event,
this way it knows the password was configured

This PR suggest other solution and focus on the SRIOV first, (it has its own container disk atm anyhow)

Reply:
I understand. thanks!